### PR TITLE
Heart ventricles 3 and arterial valves

### PR DESCRIPTION
**Pull request by rchristie**
Work in progress.

### CODE REVIEW

File: src/scaffoldmaker/utils/tracksurface.py [hunk 1]
```diff
@@ -441,10 +528,16 @@ def calculate_surface_axes(d1, d2, direction):
     ax2 in-plane normal to a and ax3 normal to the surface plane.
     Vectors all have unit magnitude.
     '''
-    ax3 = vector.normalise(vector.crossproduct3(d1, d2))
     delta_xi1, delta_xi2 = calculate_surface_delta_xi(d1, d2, direction)
     ax1 = vector.normalise([ delta_xi1*d1[c] + delta_xi2*d2[c] for c in range(3) ])
-    ax2 = vector.normalise(vector.crossproduct3(ax3, ax1))
+    ax3 = vector.crossproduct3(d1, d2)
+    mag3 = math.sqrt(ax3[0]*ax3[0] + ax3[1]*ax3[1] + ax3[2]*ax3[2])
```
[the remark this answers is not in the document]
Comment (rchristie) [reply src/scaffoldmaker/utils/tracksurface.py:534]: Fixing.

File: src/scaffoldmaker/meshtypes/meshtype_3d_heartventricles3.py [hunk 2]
```diff
@@ -0,0 +1,1106 @@
...
+            'RV proportion up LV',
+            'RV free wall thickness',
+            'RV inlet angle degrees',
+            'RV inlet cusp angle degrees',
+            'RV inlet position around LV',
+            'RV outlet angle degrees',
+            'RV outlet cusp angle degrees',
+            'RV outlet position around LV',
+            'RV width',
+            #'Use cross derivatives',
```
[the remark this answers is not in the document]
Comment (rchristie) [reply src/scaffoldmaker/meshtypes/meshtype_3d_heartventricles3.py:141]: Adding a comment for future use.

File: src/scaffoldmaker/meshtypes/meshtype_3d_heartventricles2.py [hunk 3]
```diff
@@ -749,10 +749,7 @@ def generateBaseMesh(cls, region, options):
 
                 element = mesh.createElement(elementIdentifier, elementtemplate1)
                 result2 = element.setNodesByIdentifier(eft1, nids)
-                if eft1.getNumberOfLocalScaleFactors() > 0:
-                    result3 = element.setScaleFactors(eft1, scalefactors)
-                else:
-                    result3 = 7
+                result3 = element.setScaleFactors(eft1, scalefactors) if scalefactors else None
```
Comment (mlin865) [inline src/scaffoldmaker/meshtypes/meshtype_3d_heartventricles2.py:752]: Is this supposed to be: result3 = element.setScaleFactors(eft1, scalefactors) if eft1.getNumberOfLocalScaleFactors() > 0 else None?

Comment (rchristie) [reply src/scaffoldmaker/meshtypes/meshtype_3d_heartventricles2.py:752]: I noticed that scalefactors is set to None at the start of the loop so it's more efficient and cleaner to check if it's True (i.e. not None) than call the eft function. Plus it matches the other code. 2 places allowed this change.

Comment (mlin865) [reply src/scaffoldmaker/meshtypes/meshtype_3d_heartventricles2.py:752]: ok sounds good

**Pull request merged**